### DISCUSSION TITLE
Importer: GoCardless import drops data

### DISCUSSION THREAD

**Discussion by shrippen** [firefly-iii/firefly-iii#8527] [General]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I am using the firefly importer to import data from the bank Kontist via GoCardless. Unfortunately this drops the Description Field. This is most likely caused by the data Kontist provides via GoCardless, which looks like this:

```
"transactionId": "some-data-here",
"bookingDate": "2024-02-01",
"valueDate": "2024-02-01",
"transactionAmount": {
	"amount": "-24.0",
	"currency": "EUR"
},
"creditorName": "DataData",
"additionalInformation": "Important notice for myself",
"purposeCode": "the data that should become the description field",
"internalTransactionId": "RandomString"
```

I compared the data that a different bank provides, and for that the description field is named "remittanceInformationUnstructured", which gets picked up correctly by the importer.

Would it be possible to allow the manual mapping of the gocardless data for such cases? Similar to the csv import process?
That would also allow me to store the field "additionalInformation" in the notes field of the transaction. Kontist offers the possibility to store additional text for every transaction. I use this to, for example, note which product I bought from Amazon for the corresponding transaction.

### Debug information

Debug information generated at 2024-02-05 14:27:02 for Firefly III version **v6.1.7**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.1.7 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.2 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>951, base <span>#</span>79</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Berlin + Europe/Berlin</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>info, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>de_DE + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-02-05 00:00:00 (14 hours ago)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has rules">:wrench:</span></td></tr><tr><td>Session start</td><td>2024-02-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-02-29 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>de_DE</td></tr><tr><td>User locale</td><td>de_DE</td></tr><tr><td>Locale(s) supported</td><td>de_DE.utf8: :white_check_mark:<br>de_DE.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0</td></tr></tbody></table>

**Comment by JC5** [comment]
Sorry, what you see is what you get. I can only expect banks and GoCardless to be consistent in what they put into the API. Each field has a purpose for a reason.

**Comment by shrippen** [comment]
Hmm, then I should probably contact the bank or gocardless I guess.
What field should the description be in? remittanceInformationUnstructured?

**Comment by JC5** [comment]
That field is used most often, yes.
